tls/s2n_client_cert_verify.c

```diff
@@ -48,14 +48,11 @@ int s2n_client_cert_verify_recv(struct s2n_connection *conn)
     GUARD(s2n_handshake_get_hash_state(conn, chosen_sig_scheme.hash_alg, &hash_state));
     GUARD(s2n_hash_copy(&conn->handshake.ccv_hash_copy, &hash_state));
 
-    switch (chosen_sig_scheme.sig_alg) {
-    case S2N_SIGNATURE_RSA:
-    case S2N_SIGNATURE_ECDSA:
-        GUARD(s2n_pkey_verify(&conn->secure.client_public_key, &conn->handshake.ccv_hash_copy, &signature));
-        break;
-    default:
-        S2N_ERROR(S2N_ERR_INVALID_SIGNATURE_ALGORITHM);
-    }
+    /* s2n currently only supports RSA Signatures */
+    S2N_ERROR_IF(chosen_sig_scheme.sig_alg != S2N_SIGNATURE_RSA, S2N_ERR_INVALID_SIGNATURE_ALGORITHM);
```

Comment:
This doesn't seem right at all. Why must client certs use RSA signatures and not ECDSA or RSA-PSS signatures?